apis/v1beta1/httproute_types.go

```diff
@@ -263,6 +263,56 @@ type HTTPRouteRule struct {
 	// +optional
 	// +kubebuilder:validation:MaxItems=16
 	BackendRefs []HTTPBackendRef `json:"backendRefs,omitempty"`
+
+	// Timeouts defines the timeouts that can be configured for an HTTP request.
+	//
+	// Support: Extended
+	//
+	// +optional
+	// <gateway:experimental>
+	Timeouts *HTTPRouteTimeouts `json:"timeouts,omitempty"`
+}
+
+// HTTPRouteTimeouts defines timeouts that can be configured for an HTTPRoute.
+// Timeout values are represented with Gateway API Duration formatting.
+// Specifying a zero value such as "0s" is interpreted as no timeout.
+//
+// +kubebuilder:validation:XValidation:message="backendRequest timeout cannot be longer than request timeout",rule="!(has(self.request) && has(self.backendRequest) && duration(self.request) != duration('0s') && duration(self.backendRequest) > duration(self.request))"
+type HTTPRouteTimeouts struct {
+	// Request specifies the maximum duration for a Gateway to respond to a HTTP request. If the Gateway
+	// has not been able to respond before this deadline is met, the Gateway MUST return a timeout error.
```

Comment:
nit: shorten lines to better align with the rest of the comments. Also "a HTTP" -> "an HTTP"
```suggestion
	// Request specifies the maximum duration for a Gateway to respond to an HTTP request.
	// If the Gateway has not been able to respond before this deadline is met, the Gateway
	// MUST return a timeout error.
```

Reply:
Also note that these 2 sentences have capitalized Gateway, even though they're referring to the gateway instance, not the configuration resource. This is different from other comments in this file that have gateway not capitalized. I prefer that lower case be used if not referring to the Gateway configuration resource, but more importantly it should be consistent, one way or the other.

cc @robscott WDYT?

Reply:
I agree with @frankbu here – these should be `gateway` without the capital `G`, since they refer to the controller rather than the resource.

Reply:
the capitalisation issue was fixed in e40ed42b80c4c98c3d5dd160a88a359d4bd4203d @frankbu @kflynn

Reply:
Yep, I think this is correct. I think this came from older feedback from me that suggested using `Gateway`, but I've since been convinced by @frankbu that this is the correct capitalization. Thanks for the update @SRodi!